syntax = "proto3";

package cyclecare;

service CycleCareService {
  rpc UploadVideo(stream VideoChunk) returns (UploadStatus);
  rpc StreamVideo(VideoRequest) returns (stream VideoChunk);
}

message VideoChunk {
  bytes data = 1;
  string filename = 2;
}

message UploadStatus {
  string message = 1;
  int32 code = 2;
}

message VideoRequest {
  string filename = 1;
}